dashboard/src/main/java/org/openedx/dashboard/presentation/program/ProgramFragment.kt

```diff
@@ -0,0 +1,326 @@
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.platform.ComposeView
+import androidx.compose.ui.platform.LocalConfiguration
+import androidx.compose.ui.platform.ViewCompositionStrategy
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.Dp
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.viewinterop.AndroidView
+import androidx.compose.ui.zIndex
+import androidx.core.os.bundleOf
+import androidx.fragment.app.Fragment
+import org.koin.androidx.viewmodel.ext.android.viewModel
+import org.openedx.core.UIMessage
+import org.openedx.core.presentation.catalog.CatalogWebViewScreen
+import org.openedx.core.presentation.catalog.WebViewLink
+import org.openedx.core.presentation.dialog.alert.ActionDialogFragment
+import org.openedx.core.presentation.dialog.alert.InfoDialogFragment
+import org.openedx.core.ui.ConnectionErrorView
+import org.openedx.core.ui.HandleUIMessage
+import org.openedx.core.ui.Toolbar
+import org.openedx.core.ui.WindowSize
+import org.openedx.core.ui.WindowType
+import org.openedx.core.ui.displayCutoutForLandscape
+import org.openedx.core.ui.rememberWindowSize
+import org.openedx.core.ui.statusBarsInset
+import org.openedx.core.ui.theme.OpenEdXTheme
+import org.openedx.core.ui.theme.appColors
+import org.openedx.core.ui.windowSizeValue
+import org.openedx.dashboard.R
+import org.openedx.core.R as coreR
+import org.openedx.core.presentation.catalog.WebViewLink.Authority as linkAuthority
+import org.openedx.course.R as courseR
+
+class ProgramFragment : Fragment() {
+
+    private val viewModel by viewModel<ProgramViewModel>()
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ) = ComposeView(requireContext()).apply {
+        setViewCompositionStrategy(ViewCompositionStrategy.DisposeOnViewTreeLifecycleDestroyed)
+        setContent {
+            OpenEdXTheme {
+                val windowSize = rememberWindowSize()
+                val uiMessage by viewModel.uiMessage.collectAsState(initial = null)
+                val showAlert by viewModel.showAlert.collectAsState(initial = false)
+                val showLoading by viewModel.showLoading.collectAsState(initial = true)
+                val enrollmentSuccess by viewModel.courseEnrollSuccess.collectAsState(initial = "")
+                var hasInternetConnection by remember {
+                    mutableStateOf(viewModel.hasInternetConnection)
+                }
+
+                LaunchedEffect(showAlert) {
+                    if (showAlert) {
+                        InfoDialogFragment.newInstance(
+                            title = context.getString(courseR.string.course_enrollment_error),
```

Comment:
please use stringResource fun from compose here and for others strings in this fun

Reply:
here is composabale context, it is inside setContent of the compose view